### ISSUE TITLE
训练时loss.backward出现问题了

### ISSUE THREAD

**Issue by WDYTX** [state-spaces/mamba#373]
![image](https://github.com/state-spaces/mamba/assets/98384255/0890ffdd-a158-42ac-98d5-0b69a1c20023)

这该如何解决啊

**Comment by huashui-gif** [comment]
I have a same error : when I train use loss.backward() 
llvm::SmallVector<unsigned int> mlir::triton::gpu::getElemsPerThread(mlir::Attribute, llvm::ArrayRef<long int>, mlir::Type): Assertion `0 && "getElemsPerThread not implemented"' failed.

**Comment by WDYTX** [comment]
我把ssd_combined.py中的这行代码dA_cs_k = tl.load(dA_cumsum_ptrs, mask=offs_k < K_MAX - k, other=0.0).to(tl.float32)改成了这个dA_cs_k = tl.load(dA_cumsum_ptrs, mask=offs_k < K_MAX - k, other=-1e6).to(tl.float32)就可以跑起来了

**Comment by xueaa** [comment]
> 我有一个同样的错误：当我训练使用loss.backward（） llvm：：SmallVector mlir：：triton：：gpu：：getElemsPerThread（mlir：：Attribute， llvm：：ArrayRef， mlir：：Type）： 断言'0 && “getElemsPerThread not implemented”' failed.

Have you solved this issue?

**Comment by catalpaaa** [comment]
i think its cause by #352

**Comment by zllxot** [comment]
> I have a same error : when I train use loss.backward() llvm::SmallVector mlir::triton::gpu::getElemsPerThread(mlir::Attribute, llvm::ArrayRef, mlir::Type): Assertion `0 && "getElemsPerThread not implemented"' failed.

same error

**Comment by ihavekonwnothing** [comment]
> I have a same error : when I train use loss.backward() llvm::SmallVector mlir::triton::gpu::getElemsPerThread(mlir::Attribute, llvm::ArrayRef, mlir::Type): Assertion `0 && "getElemsPerThread not implemented"' failed.

same error

**Comment by zllxot** [comment]
> > I have a same error : when I train use loss.backward() llvm::SmallVector mlir::triton::gpu::getElemsPerThread(mlir::Attribute, llvm::ArrayRef, mlir::Type): Assertion `0 && "getElemsPerThread not implemented"' failed.
> 
> same error

After updating to mamba-ssm 2.2.2 version, the problem was solved
